src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageCollection{T}.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.UpgradeAssistant.Packages;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
+{
+    internal class PackageCollection<T> : IDependencyCollection<T>
```

Comment:
Nit-pick: `DependencyCollection` might be a better name since this is used both for package dependencies and assembly/framework references.